src/idl_gen_cpp.cpp

```diff
@@ -513,15 +521,32 @@ static void GenTable(const Parser &parser, StructDef &struct_def,
 }
 
 static void GenPadding(const FieldDef &field,
-                       const std::function<void (int bits)> &f) {
+            std::string &code,
+            int &padding_id,
+            const std::function<void (int bits, std::string &code, int &padding_id)> &f) {
   if (field.padding) {
     for (int i = 0; i < 4; i++)
       if (static_cast<int>(field.padding) & (1 << i))
-        f((1 << i) * 8);
+        f((1 << i) * 8, code, padding_id);
     assert(!(field.padding & ~0xF));
   }
 }
 
+static void PaddingDefinition(int bits, std::string &code, int &padding_id) {
+  code += "  int" + NumToString(bits) +
+          "_t __padding" + NumToString(padding_id++) + ";\n";
+}
+
+static void PaddingDeclaration(int bits, std::string &code, int &padding_id) {
```

Comment:
It seemed a bit silly to use the struct Padding::Method approach earlier, so moved these out.